src/views/recordTable.component.tsx

```diff
@@ -31,9 +31,12 @@ const RecordTable = React.memo((): React.ReactElement => {
     useAvailableColumns();
 
   const columnStates = useAppSelector(selectColumnStates);
+
+  const { data: allChannels } = useChannels(); // just for use with hiddenColumns, not passed anywhere else
```

Comment:
I don't like essentially querying for the same data twice (via `useAvailableColumns` and `useChannels`) just for passing into `selectHiddenChannels`. Since all `selectHiddenChannels` is is a `string[]` and we essentially are just comparing "channel id"/`systemName` - is it possible to bring back `selectHiddenColumns` - i.e. the selector takes in `availableColumns` and uses that to filter the `string[]`? We can have two selectors `selectHiddenColumns` & `selectHiddenChannels` if necessary.

Reply:
So transform selectHiddenChannels back into what it was before where it was returning an array of react-table Columns and have columnCheckboxes use that as before? Happy to do it for ease of use in the code like you say, just want to make sure I have your meaning right

Reply:
`selectHiddenColumns` always returned a `string[]` - the difference is what argument it expects. It makes sense to me for it to accept `Columns` as that's what we query for in `recordTable` (via `useAvailableColumns`). Creating extra selectors is pretty easy - if we ever needed a `selectHiddenChannels` selector that accepts `FullChannelMetadata[]` instead we could create an extra selector rather than refactor `selectHiddenColumns`